sdk/metric/manual_reader.go

```diff
@@ -103,7 +120,19 @@ func (mr *manualReader) Collect(ctx context.Context) (metricdata.ResourceMetrics
 		return metricdata.ResourceMetrics{}, err
 	}
 
-	return ph.produce(ctx)
+	rm, err := ph.produce(ctx)
+	if err != nil {
```

Comment:
It seems that 
- the failed calling produce() of SDK  producer will result that the external producers' Produce() not be collected.
- the failed calling produce() of external producers will not affect the SDK producer Produce().

Is this logic is by design? The collect of SDK producer has higher priority than external producers?

Reply:
The external producers only provide scope metrics, and not a resource.  That has the advantage of not requiring bridges to be involved in the resource portion of OTel (e.g. with discovery, etc.).

But that means that if the SDK fails, we won't have a resource that we can use to produce a "complete" batch of metrics.  Rather than returning metrics with just a scope, we just treat SDK errors as fatal to the entire collection.

Alternatively, we could make the SDK return partial errors as well if a callback fails.